FPGA CI shield: Add a watchdog timing test

fkjagodzinski opened the pull request:
### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->
Replace the `tests-mbed_hal-watchdog_timing` with ~`tests-mbed_hal_fpga_ci_test_shield-watchdog_timing`~ `tests-mbed_timing_fpga_ci_test_shield-watchdog`.

The timeout measurement from the original test is based on heartbeat messages sent to a host PC via the greentea serial. This method is fairly inaccurate because the last timestamp received by the host will always be before the expected reset time. Thanks to `IOMetrics` feature of our new FPGA CI test shield, the watchdog timeout measurement boils down to counting the falling edges of the heartbeat signal. The pin used to output this signal is added to a list of restricted pins. This guarantees that no other signals (i.e. FPGA control) will interfere.

### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [x] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

@mprse, @maciejbocianski, @jamesbeyond

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

fkjagodzinski committed:
Test: FPGA shield: Add a Watchdog-timing test

Use the FPGA IOMetrics feature to verify the watchdog timeout accuracy.

mprse reviewed (changes requested):
The test is okay, but I think we should decide:
- where to put the timing tests,
- do we want to remove old tests (which use host pc to count elapsed time)

The same problem I have here: PR https://github.com/ARMmbed/mbed-os/pull/11147.

Currently, all FPGA tests are in the following location:
`TESTS\mbed_hal_fpga_ci_test_shield`
 
I'm not sure if this is a good idea to put alternative timing tests in the same location together with peripheral fpga tests. My proposition is to create an additional folder for the timing tests. Something like:
`TESTS\mbed_timing_fpga_ci_test_shield`.

Alternatively, we may also break the current convention and add FPGA tests close too non-FPGA versions:
`TESTS\mbedmicro-rtos-mbed\basic`
`TESTS\mbedmicro-rtos-mbed\basic_fpga`.

I think we should not remove the original tests since they might be valuable to users who do not have FPGA shields. On CI we may run only one test version depending if FPGA is supported for the board.

fkjagodzinski committed:
Test: FPGA shield: Add directory for timing tests

Create a directory for all timing tests that make use of the FPGA
shield, regardless of the component they test.

Move the FPGA-Watchdog-timing test.
